core/src/lib.rs

```diff
@@ -77,3 +80,5 @@ pub use parking_lot::{ParkResult, UnparkResult, RequeueOp, UnparkToken, ParkToke
 pub use parking_lot::{DEFAULT_UNPARK_TOKEN, DEFAULT_PARK_TOKEN};
 pub use parking_lot::{park, unpark_one, unpark_all, unpark_requeue, unpark_filter};
 pub use spinwait::SpinWait;
+#[cfg(feature = "deadlock_detection")]
+pub use parking_lot::{register_resource, deregister_resource, check_deadlock};
```

Comment:
I think the names `register_resource` and `deregister_resource` aren't very informative: it isn't immediately clear that this is related to deadlock detection. Maybe you could put these in their own submodule `parking_lot_core::deadlock_detector`.

Reply:
I have no idea how I came up with these, resources is a textbook parlance. I'll think of something.

Reply:
I think that with the added submodule and some naming changes, it becomes clearer:

- `parking_lot_core::deadlock_detector::acquire_resource`
- `parking_lot_core::deadlock_detector::release_resource`